app/components/primer/tab_nav_component.rb

```diff
@@ -132,19 +135,23 @@ def initialize(label:, with_panel: false, body_arguments: {}, wrapper_arguments:
       @body_arguments = body_arguments
       @wrapper_arguments = wrapper_arguments
 
-      @system_arguments[:tag] = :div
+      @system_arguments[:tag] = navigation_tag(with_panel)
```

Comment:
I don't want to bloat this PR so in a follow up, I will change this so consumers have option to render page navigation links NOT in a `nav`.

Currently page navigation always gets rendered in a `nav`.`nav` may not be appropriate when the links aren't for main navigation. I noticed the GitHub explore page uses `TabNav` to list repository links, and these are showing up as navigation landmark which we probably don't want here. Navigation landmark should be reserved for primary/secondary navigation.

<img width="527" alt="NVDA landmark menu for github explore page showing multiple navigations" src="https://user-images.githubusercontent.com/16447748/124020572-4bd1cc00-d99f-11eb-84c0-c9cce28b07d6.png">

We should have option to render `div` instead of `nav so the markup is:
```
<div> 
  <ul>
    <li>
    <li>
  </ul>
</div>
```